nupic/research/frameworks/pytorch/datasets/gsc_factory.py

```diff
@@ -46,7 +46,9 @@ def preprocessed_gsc(root, train=True, download=True):
     will be reused.
 
     :param root: directory to store or load downloaded data
-    :param train: whether to load train of validation data
+    :param train: whether to load train or test data
```

Comment:
Having 3 boolean parameters here is really confusing. In addition, to test on the test set you have to set all three to False, which is quite strange. 

There are 8 possible choices - are all valid? (Clearly not, there are only 4 possible outcomes.) There is no need to maintain backward compatibility here, so feel free to use a more natural version.